KubeArmor/BPF/shared.h

```diff
@@ -508,22 +516,64 @@ static inline int match_and_enforce_path_hooks(struct path *f_path, u32 id,
   }
 
 decision:
-
-  if (id == dpath) { // Path Hooks
+  if (id == dpath) { // Path Hooks  
     if (match) {
       if (val && (val->filemask & RULE_OWNER)) {
-        if (!is_owner_path(f_path->dentry)) {
-          retval = -EPERM;
+        struct dentry *dent ;
+        if(eventID  == _FILE_MKNOD || eventID == _FILE_MKDIR){
+          dent = BPF_CORE_READ(f_path , dentry , d_parent);
```

Comment:
Let's comment the reasoning here.

MKNOD and MKDIR would always have current user as the owner so we need to check the parent dentry's owner to enforcer owner only